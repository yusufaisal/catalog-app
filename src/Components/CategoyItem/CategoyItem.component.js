import * as React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';

import styles from './CategoyItem.component.styles';

const _renderName = (props) => (
  <Text style={styles.name}>{props.name}</Text>
);

const CategoyItem = (props) => (
  <TouchableOpacity>
    <View style={styles.container}>
      {_renderName(props)}
    </View>
  </TouchableOpacity>
);

export default CategoyItem;