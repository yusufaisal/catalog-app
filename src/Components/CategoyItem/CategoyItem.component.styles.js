const shadow = {
  shadowColor: '#171717',
  shadowOffset: {width: 1, height: 1},
  shadowOpacity: 0.1,
  shadowRadius: 3,
}

export default {
  container: {
    backgroundColor: '#FFFFFF',
    borderColor: '#ECECEC',
    borderWidth: 1,
    borderRadius: 6,
    paddingHorizontal: 12,
    paddingVertical: 12,
    marginHorizontal: 5,
    marginVertical: 5,
    ...shadow
  },
  name: {
    fontWeight: '500',
    fontSize: 16
  }
};