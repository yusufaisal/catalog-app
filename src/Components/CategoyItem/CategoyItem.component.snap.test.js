import React from 'react';
import renderer from 'react-test-renderer';

import CategoyItem from './CategoyItem.component';

describe('CategoyItem Component', () => {
  it('should render category item properly', () => {
    const item = {
      "id": "11",
      "name": "T-Shirts"
    }
    const tree = renderer
      .create(<CategoyItem name={item.name} key={item.id}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});