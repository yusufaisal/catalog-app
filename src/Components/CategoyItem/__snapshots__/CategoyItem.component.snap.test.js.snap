// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoyItem Component should render category item properly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#ECECEC",
        "borderRadius": 6,
        "borderWidth": 1,
        "marginHorizontal": 5,
        "marginVertical": 5,
        "paddingHorizontal": 12,
        "paddingVertical": 12,
        "shadowColor": "#171717",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      T-Shirts
    </Text>
  </View>
</View>
`;
